gfx/drivers/gl.c

```diff
@@ -379,8 +379,7 @@ static bool gl_shader_init(gl_t *gl, const gfx_ctx_driver_t *ctx_driver,
 {
    video_shader_ctx_init_t init_data;
    settings_t *settings            = config_get_ptr();
-   const char *shader_path         = (settings->bools.video_shader_enable
-         && *settings->paths.path_shader) ? settings->paths.path_shader : NULL;
+   const char *shader_path         = retroarch_get_shader_preset();
```

Comment:
@bparker06 should I strdup here? it should only happen at startup and the only time this value changes is at deinit but maybe that would be safer?

I'm doing this on every driver that supports shaders now so it's important to know

Reply:
When I click "view changes" it says it couldn't find the commit for some reason. Anyway, I don't see why it even lets you compile `retroarch_get_shader_preset` because returning `char*` for a string literal (e.g. `return ""` in the function's definition) is invalid and will crash if you try to modify it... and I'm also not sure what strdup("") might do on various platforms if it tried to do that.

Those things aside, I can't really answer your question. I don't know what may or may not be going on that would require this string to either be modified or rely on its value not being changed while running, but the fact that you are currently assigning it to `const char*` makes me assume it does not need to be modified (and even if you did strdup, you still can't modify it if it's `const` anyway). I guess one way to see more in-depth as to what's going on is to set watchpoints in gdb to see where all this string is being accessed and what is being done with it.

Reply:
Thanks for the feedback, I'll fix the return "", I was actually doing a return null but some drivers don't handle it properly (despite it being the original return value)

Reply:
Fixed :)